local-volume/provisioner/pkg/types/types.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/cache"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/util"
+)
+
+const (
+	AnnProvisionedBy = "pv.kubernetes.io/provisioned-by"
+)
+
+type UserConfig struct {
+	// Name of the node this is running on
+	// TODO: Need to resolve nodename vs hostname
+	NodeName string
+	// The hostpath directory
+	HostDir string
```

Comment:
my point is we can support multiple hostdir, e.g.

/mnt/local-volumes/class1
/mnt/local-volumes/class2

/data/local-volumes/class3
/data/local-volumes/class4

so we have discoverymap = { "sc1": {hostpath: "/mnt/local-volumes/", relativePaths: []string{"class3"}}

just a rough idea, i think this can be deferred to dynamic setting.